marlowe-actus/src/Language/Marlowe/ACTUS/Model/INIT/StateInitialization.hs

```diff
@@ -20,11 +20,11 @@ shift = applyBDCWithCfg
 inititializeState :: ContractTerms -> ContractState
 inititializeState terms@ContractTerms {..} =
     let t0         = ct_SD
-        fpSchedule = fromMaybe [shift scfg t0] $ schedule FP terms
-        ipSchedule = fromMaybe [shift scfg t0] $ schedule IP terms
-        tminus     = calculationDay $ sup ipSchedule t0
-        tfp_minus  = calculationDay $ sup fpSchedule t0
-        tfp_plus   = calculationDay $ inf fpSchedule t0
+        fpSchedule         = schedule FP terms
+        tfp_minus          = fromMaybe t0 $ calculationDay <$> ((\sc -> sup sc t0) =<< fpSchedule)
+        tfp_plus           = fromMaybe t0 $ calculationDay <$> ((\sc -> inf sc t0) =<< fpSchedule)
+        ipSchedule         = schedule IP terms
+        tminus             = fromMaybe t0 $ calculationDay <$> ((\sc -> sup sc t0) =<< ipSchedule)
```

Comment:
No tfpminus is previous FP (fee payment) event. tminus is just previous event - it might not be FP.

Reply:
ah, I didn't see `ipSchedule` vs `fpSchedule`. Which suggests to me that this whole thing should be a function, then I can see the two parameters being different. `f sup fpSchedule`

Reply:
This equation is a part of their spec, that's why I'm affraid to move it too far idk. 
I'm not sure internal function would help given that it would only generalize two cases

Reply:
ah ok, cool.